document/src/main/kotlin/com/ritense/document/domain/EveritSchemaAllowsProperty.kt

```diff
@@ -0,0 +1,198 @@
+/*
+ * Copyright 2015-2023 Ritense BV, the Netherlands.
+ *
+ * Licensed under EUPL, Version 1.2 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://joinup.ec.europa.eu/collection/eupl/eupl-text-eupl-12
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" basis,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.ritense.document.domain
+
+import org.everit.json.schema.ArraySchema
+import org.everit.json.schema.CombinedSchema
+import org.everit.json.schema.JSONPointer
+import org.everit.json.schema.ObjectSchema
+import org.everit.json.schema.ReferenceSchema
+import org.everit.json.schema.Schema
+import org.everit.json.schema.ValidationException
+import org.everit.json.schema.regexp.Regexp
+
+/** This method is a copy from org.everit.json.schema.Schema.definesProperty(.) but returns true when the schema allows additionalProperties */
+fun Schema.allowsProperty(field: String): Boolean {
```

Comment:
Can you add a test to validate this method works as intended after the changes made here?